2.2.2 backports 1

gsmet opened the pull request:
Please don't merge, I will merge it myself.

stuartwdouglas committed:
Add some substitutions for Brotli and Zstd

(cherry picked from commit fb4c1cbb92ad588bd48cec3d07a223ddfec73a91)

Sanne committed:
Upgrade the Oracle JDBC driver to benefit from JDK11 baseline

(cherry picked from commit 766dd619bdf614870576a2bfef5399d1dae3c3ab)

geoand committed:
Add the ability for extensions to exclude GraalVM config from jars

Resolves: #17482
(cherry picked from commit f36724b75ffcedc4c8c7cdb05a2c449c2cd58080)

glefloch committed:
Remove hard coded dependency version in ConditionalDependenciesTest

(cherry picked from commit 93bf9f51f2882e0a222fc3ac5e7e0f7b8686a779)

sberyozkin committed:
Update OIDC TokenStateManager to return Uni

(cherry picked from commit e4d2bcfe681928e87e7309bb3a5c7c9c6d35ef7a)

stuartwdouglas committed:
Update Dev UI with the correct state when resuming

Fixes #19777

(cherry picked from commit 25ab92601ba13d2fe083d48578028e69d4ec2bc8)

glefloch committed:
Set default jvm-target in kotlin sample project

(cherry picked from commit a77eadb930cb39d3e07a0d009e6a04ba2667246d)

manusa committed:
fix: K8s Service can be re-applied

Signed-off-by: Marc Nuri <marc@marcnuri.com>
(cherry picked from commit 7142e7913f0f476889b7353432d339ca14b2ad12)

mkouba committed:
Codestarts - gRPC - add index.html

- resolves #19642

(cherry picked from commit 57d2f7cfd67c7690b7aba46b24360c43573fa52d)

Sanne committed:
Make sure the --exclude-config arguments precede the -jar arguments

(cherry picked from commit 58e48e0594f43992a89209826da98348f41ac84e)

cescoffier committed:
Update Vert.x version to 4.1.3 and Mutiny bindings to 2.13.0

(cherry picked from commit 2d9e9dc35b0074035df6093a8e7c05b3fb861f03)

famod committed:
Fix NoSuchElementException on `@Before/AfterAll` with TestInfo

Fixes #19800

(cherry picked from commit 6ac8ad3884bfee5915c42f14b0bb7bfff8d30baf)

ebullient committed:
CLI: docs and help for -P

(cherry picked from commit f78e00ace9d9696cdebd499817a7cb70023ccd55)

glefloch committed:
Do not propagate java compiler argument in kotlin compilation provider

(cherry picked from commit d0d2d57c686ea156c03eb7ea865ebb6be70ed6d2)

stuartwdouglas committed:
Fix some tests the broke due to default change

The max attribute size changed in vert.x

(cherry picked from commit a3d5b90cc5fbe17ae79864c0f3868369715d8d46)

cescoffier committed:
Update the status of the mutiny, AMQP connector and gRPC extensions to "stable".

(cherry picked from commit b1d3c9d961334bd88a61ee91eef59305d5ae3960)

stuartwdouglas committed:
Make sure KC Dev UI page persists after reload

(cherry picked from commit 03647998696e6a7e202b7e3a07c0ecb2883b5f1e)

geoand committed:
Fix @NoCache without fields handling in RESTEasy Reactive

Fixes: #19822
(cherry picked from commit 07db37f530aa9450874f2e96b31505ab3e39e622)

geoand committed:
Mark quarkus-container-image-openshift as stable

(cherry picked from commit 12a882fd3a8ebbe136360d20511037336b585a6e)

michalszynkiewicz committed:
Rest Client Reactive: Multipart file name set to actual filename for path and file

fixes #19805

(cherry picked from commit 55c8f016384ffaf348154498393c98793b5e972c)

jmartisk committed:
SmallRye GraphQL 1.3.2

(cherry picked from commit 40ced5eed45d9ee94e42df8e99b46fa9043410fa)

geoand committed:
Provide actionable error message when RESTEasy Reactive can't pick Provider constructor

Relates to: #19750

Co-authored-by: George Gastaldi <gegastaldi@gmail.com>
(cherry picked from commit 12ca7ebfbf4d170f6775447cb2658db39c7c10f1)

manusa committed:
Bump kubernetes-client-bom from 5.7.0 to 5.7.2

Signed-off-by: Marc Nuri <marc@marcnuri.com>
(cherry picked from commit 679561e16faa0da7a9e8031716652b19981b2130)

gastaldi committed:
Marking Amazon extensions as stable

(cherry picked from commit bac9ee06145d3b1595f1a38eddfcb465820015a1)

geoand committed:
Take @TestSecurity(authorizationEnabled = false) into account for RESTEasy Reactive

Relates to: #19834

(cherry picked from commit 07f9e8741d35016df1727ffd834412b8d2bb1b15)

geoand committed:
Don't fail Hibernate Validator when no RESTEasy Reactive request is in flight

Relates to: #19834

(cherry picked from commit ee298ecdfdd258b6af47fe2b8daa0d3f602031c1)

phillip-kruger committed:
Dev UI Test result screen make sure dark background use light test (Fix #19374)
Signed-off-by:Phillip Kruger <phillip.kruger@gmail.com>

(cherry picked from commit c83c608d31c719968549162bb3ba81fbbf8f126f)

phillip-kruger committed:
Dev UI Keykloak - Make the logout button visible
Signed-off-by:Phillip Kruger <phillip.kruger@gmail.com>

(cherry picked from commit fab4d7045544a3170916cb316028c036b97ae405)

michalszynkiewicz committed:
Rest Client Reactive: improve error on invalid subresource return type

(cherry picked from commit 0b18024b6f10fea094688f37540ad03c90ab5a9f)

stuartwdouglas committed:
Clean up hang detection on failed start

This can cause a false positive in the vertx-http-deployment tests where
a continuous test fails to start, then spits out a hang detection
warning a minute later.

(cherry picked from commit ebddbc3bc1d22753c23d198ed7d8df0ffa6315ab)

aloubyansky committed:
Experimental Maven goal to check for Quarkus platform and extension updates

Co-authored-by: Guillaume Smet <guillaume.smet@gmail.com>
(cherry picked from commit 05b4d363ade67cb76dd5d46760f8e32a5fa7192b)

iocanel committed:
fix: KubernetesClientErrorHandler always throws a RuntimeException

(cherry picked from commit 57b1ffd5135de82017fe6398e87c67aa09f82c02)

barreiro committed:
Expose Agroal feature: idle validation

(cherry picked from commit 3f37b7f8411bd87f450f83432fbcb99187fce12b)

kdubb committed:
Update JNA to work with Apple M1

(cherry picked from commit df57f2d26ea9a4a3bf375aa2656a92c6ed01f2ac)

kenfinnigan committed:
Fix RESTEasy classic server templating for observability

- Fixes #18251
- Add test to Micrometer integration test. Verified not a problem with RESTEasy Reactive
- Add test to OpenTelemetry integration test to very fix for RESTEasy Classic

(cherry picked from commit d57ddba8b3c19a78ad0dff9663c37b5c30d9b012)

Karm committed:
LambdaContainerHandlerSubstitution class must be marked final

(cherry picked from commit d35a89a40e5a1036de54697b45c5bad21cc7366c)

geoand committed:
Ensure that eager security handling works in native mode

Fixes: #19878
(cherry picked from commit 690210e6db847c84736d0cf63c255958943e7c40)

geoand committed:
Fix use of OpenTelemetry OTLP exporter in native mode

Essentially the fact that ClientTracingFilter was using a static
field for TextMapPropagator was causing GraalVM to initialize
the entire telemetry infrastructure at build time

Fixes: #19877
(cherry picked from commit d142b2fe4f201e67583391c2c14ebde110f219db)

mkouba committed:
ArC - make it possible to identify additional bean archives

(cherry picked from commit c91431c9b1eea471d62da10a819be41e279dc7d9)

mkouba committed:
gRPC - register additional bean archives

- an application archive that contains a MutinyBean implementor is an
additional bean archive
- resolves #19864

(cherry picked from commit 13bc4d27509dad67c6ef4fcfb4148ab23d802214)

glefloch committed:
Fix gradle deployment classpath leak

(cherry picked from commit 66126ab4a48f8f0413bf269c21669e5324f082d7)

geoand committed:
Fix issue with skip predicates in scheduled methods

Before this fix, the implementations where actually
eligible for bean removal

Fixes: #19900
(cherry picked from commit 361494784de7ae7aa73e755be57d3a417113f2da)

Sanne committed:
Upgrade to Oracle JDBC driver v 21.3.0.0 and import its BOM

(cherry picked from commit f7b48a9871ad46d19cb93faa112dfe79b65988fc)

Sanne committed:
Override the Oracle JDBC automatic native-image metadata and re-implement it explicitly

(cherry picked from commit a7f87e266f8f68028fd7b0c4d51e1d4038ba59a2)

aloubyansky committed:
Include relevant upstream catalogs even if their versions aren't recommended for new projects

Moved io.quarkus.maven.StreamCoords to io.quarkus.registry.catalog.PlatformStreamCoords

(cherry picked from commit 9a15b82a7047d555f3bfdfb5530f046929539b79)

EricWittmann committed:
Fixed a broken link in a guide

(cherry picked from commit 793ffd7300c1ac4a732fc8cdf5e995d14aa95b8a)

manaswinidas committed:
Fix broken link for config-reference

(cherry picked from commit 8798bd6a2c9002d4c8f8ec130adb160d4a6097e1)

famod committed:
Fix combination of multiple `@Nested` tests with root lifecycle methods

Fixes #19843

(cherry picked from commit 34391e670daa47ae9491d64e4c57d411f77e986a)

famod committed:
Set default jvm-target in basic kotlin sample project

(cherry picked from commit 9d7292d0c58714419f94744970bc00123df8f955)

manaswinidas committed:
Add instructions to access App passwords page

(cherry picked from commit 80894adb77c298557b4834989aaf9dcb53ca58bd)

Sanne committed:
Introduce a build item to control flag --allow-incomplete-classpath of native-image

(cherry picked from commit 5fab70f27a5380ec85db3da373930445634d75a4)

sberyozkin committed:
Mark oidc-client and oidc-client-filter as stable and update descriptions for all OIDC extensions

(cherry picked from commit 61ef8d84a29fb774619fb2a2fb3193400bd6f0e7)

radcortez committed:
Revert "Register @ConfigMappings directly into the Config builder"

This reverts commit fd4b3028

(cherry picked from commit 3504a54217e64a482b88c1345106a01a86b9f005)

gsmet committed:
Fix build report archive name for Initial JDK 11 build

(cherry picked from commit 68be98e02d00ab7b5f10bb83fcff11abaa7cf4c0)

gsmet committed:
CI - Always include a file at the root to get the full hierarchy

The upload-artifact action takes the common ancestor as the root path of
the archive which will lead to issues when we have only one file.

Always uploading a file from the root of the repository should fix this.

(cherry picked from commit 370d94426dd560618465f4682c05f2c77a12b889)

sii-ggrebert committed:
add mailer-deployment to bom

(cherry picked from commit ab82e7ae303c15908f54616ad87f61da7457edc6)

famod commented:
Looking at
```
2021-09-06T15:48:46.2416937Z [INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 36.939 s - in io.quarkus.gradle.devmode.BasicJavaPlatformModuleDevModeTest
2021-09-06T15:48:46.2420421Z [INFO] Running io.quarkus.gradle.devmode.BasicKotlinApplicationModuleDevModeTest
2021-09-06T15:49:48.0619555Z BELOW IS THE CAPTURED LOGGING OF THE FAILED GRADLE TEST PROJECT BUILD
2021-09-06T15:49:48.0658927Z $ D:\a\quarkus\quarkus\integration-tests\gradle\gradlew.bat --no-daemon -D=maven.repo.local=C:\Users\runneradmin\.m2\repository --stacktrace clean quarkusDev -s
2021-09-06T15:49:48.0840477Z To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/6.9/userguide/gradle_daemon.html#sec:disabling_the_daemon.
2021-09-06T15:49:48.0842939Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 61.806 s <<< FAILURE! - in io.quarkus.gradle.devmode.BasicKotlinApplicationModuleDevModeTest
2021-09-06T15:49:48.1158709Z Daemon will be stopped at the end of the build 
2021-09-06T15:49:48.1351250Z [ERROR] io.quarkus.gradle.devmode.BasicKotlinApplicationModuleDevModeTest.main  Time elapsed: 61.799 s  <<< ERROR!
```
I wonder whether this is due to #19584 not having been backported as well. /cc @glefloch

gsmet commented:
I have no idea but tbh I'm by far more concerned about the other failures :/

stuartwdouglas committed:
Fix log assertions

This should not prevent Quarkus from shutting down

(cherry picked from commit 31138b0896045aa53099422d0185a45ad06b6c2d)

stuartwdouglas committed:
SCI's can't add websocket endpoints

Fixes #19781

(cherry picked from commit fb37943764b5f700b3e2a2127de23bc6adee271f)

stuartwdouglas committed:
Fix continuous testing race

Files to watch must be updated before the listeners are notified,
otherwise the test may modify the config file before the file set has
been updated, and the tests will not be run.

(cherry picked from commit 18852338c81eb38cc9cc22ab98a36a771b10fddb)

gsmet committed:
Force jvm-target 11 for BasicKotlinApplicationModuleDevModeTest
